apps/mobile/src/features/send/components/recipient/v2/recipient-selector/recipient-selector.tsx

```diff
@@ -38,27 +46,41 @@ export function RecipientSelector({
     selectedAccount,
     assetInfo,
   });
-
-  const showLoadingIndicator = searchTerm.length > 0 && recipientSuggestions.isFetching;
+  const isPerformingSearch = normalizeSearchTerm(searchTerm).length > 0;
+  const isBnsLookup = isBnsLookupCandidate(normalizeSearchTerm(searchTerm));
 
   return (
     <>
-      <Box px="5" py="5" pb="4">
+      <RecipientSelectorHeader>
         <RecipientInput value={searchTerm} onChange={setSearchTerm} />
-      </Box>
-      {showLoadingIndicator && <SendFormLoadingSpinner />}
-      {!showLoadingIndicator && recipientSuggestions.isSuccess && (
-        <BottomSheetSectionList
-          sections={recipientSuggestions.data}
-          keyboardShouldPersistTaps="always"
-          keyExtractor={item => item.id}
-          stickySectionHeadersEnabled={false}
-          renderSectionHeader={({ section }) => <RecipientSelectorSectionHeader id={section.id} />}
-          renderItem={({ item }) => (
-            <RecipientSelectorItem entry={item} onSelect={onSelectAddress} />
-          )}
-        />
-      )}
+      </RecipientSelectorHeader>
+      {matchSuggestionsResult({
```

Comment:
This is a cool syntax but any reason not to just just a component here? `<SuggestionsResult` 

I guess it's a nice pattern passing in the object props?

Reply:
Almost no strong reason really--experimenting with different variations of this for now to pinpoint main use cases and make a single reusable matcher we can use in most places. And it will likely be a component vs. a helper function like this one. There's a lot to cover with this, I'll share some ideas on the call today.